breakpad-symbols/src/sym_file/types.rs

```diff
@@ -46,6 +65,8 @@ pub struct Function {
     pub name: String,
     /// Source line information for this function.
     pub lines: RangeMap<u64, SourceLine>,
+    /// Inlinee information for this function, sorted by (depth, address).
+    pub inlinees: Vec<Inlinee>,
```

Comment:
I guess it probably doesn't matter because sorting is sorting, but it feels deeply bizarre to sort by depth more significantly than address? Surely address is the "significant" thing and depth is the "less significant" thing? Does this do something magical?

Reply:
Haha yes, this definitely deserves a comment. It threw me off too when I implemented the binary search over inlines in the addr2line crate.

Here's an example where sorting by `(address, depth)` would take you off-track:

```
inlinees: [
  { address: 0x0, size: 0x10, depth: 0 },
  { address: 0x0, size: 0x2, depth: 1 },
  { address: 0x6, size: 0x2, depth: 1 },
]

Look up { address: 0x7, depth: 0 }
```

In that example, if you sort by `(address, depth)`, your `Err(i) => pick item at i - 1` case will make you end up at an inlinee with depth 1.

Reply:
Ohh it's like a flame graph, the lower frames can actually span a wider range. Interesting. Ok so really this is logically a `Vec<Vec<Inlinee>>`, where each element of the outer vec is "the different depths" and the inner vec is "all the frames at this depth". You're really only trying to binary search the inner vec, but you've opted to flatten everything so the binary search can still accidentally peek into the next/previous depth, but you have guards in place to catch that.

Reply:
That's exactly right! Those are great ways to conceptualize it which I hadn't thought of before. I had treated inline information as "a nested structure covering address ranges", but the flame graph comparison works really well, thanks for that idea.

I've tweaked the comment to say that this can be thought of as "one vec per depth" (but concatenated into one), and to mention the flame graph conceptual model.